hnn_core/network.py

```diff
@@ -401,6 +556,150 @@ def _get_src_type_and_pos(self, gid):
 
         return src_type, src_pos, src_type in self.cellname_list
 
+    def _extract_bias_specs_from_hnn_params(self):
+        """Create 'bias specification' dicts from saved parameters
+        """
+        bias_specs = {'tonic': {}}  # currently only 'tonic' biases known
+        for cellname in self.cellname_list:
+            short_name = _short_name(cellname)
+            is_tonic_present = [f'Itonic_{p}_{short_name}_soma' in
+                                self.params for p in ['A', 't0', 'T']]
+            if any(is_tonic_present):
+                if not all(is_tonic_present):
+                    raise ValueError(
+                        f'Tonic input must have the amplitude, '
+                        f'start time and end time specified. One '
+                        f'or more parameter may be missing for '
+                        f'cell type {cellname}')
+                bias_specs['tonic'][cellname] = {
+                    'amplitude': self.params[f'Itonic_A_{short_name}_soma'],
+                    't0': self.params[f'Itonic_t0_{short_name}_soma'],
+                    'T': self.params[f'Itonic_T_{short_name}_soma']
+                }
+        return(bias_specs)
+
+    def _extract_drive_specs_from_hnn_params(self):
```

Comment:
same comment, should be a function. And perhaps, these kind of of functions could be put in a separate file `legacy_drives.py` ?